spec/requests/chat_controller_spec.rb

```diff
@@ -630,4 +630,64 @@
       expect(response.status).to eq(200)
     end
   end
+
+  describe "#quote_messages" do
+    fab!(:channel) { Fabricate(:chat_channel, chatable: category, name: "Cool Chat") }
+    let(:user2) { Fabricate(:user) }
+    let(:message1) { Fabricate(:chat_message, user: user, chat_channel: channel, message: "an extremely insightful response :)") }
+    let(:message2) { Fabricate(:chat_message, user: user2, chat_channel: channel, message: "says you!") }
+    let(:message3) { Fabricate(:chat_message, user: user, chat_channel: channel, message: "aw :(") }
+
+    it "returns a 403 if the user can't chat" do
+      SiteSetting.chat_allowed_groups = nil
+      sign_in(user)
+      post "/chat/#{channel.id}/quote.json", params: { message_ids: [message1.id, message2.id, message3.id] }
+      expect(response.status).to eq(403)
+    end
+
+    it "returns a 403 if the user can't see the channel" do
+      category.update(read_restricted: true)
+      sign_in(user)
+      post "/chat/#{channel.id}/quote.json", params: { message_ids: [message1.id, message2.id, message3.id] }
+      expect(response.status).to eq(403)
+    end
+
+    it "returns a 403 if the channel is a DM channel" do
+      sign_in(user)
+      dm_channel_chatable = Fabricate(:direct_message_channel, users: [user, user2])
+      dm_channel = Fabricate(:chat_channel, chatable: dm_channel_chatable)
+      message1.update(chat_channel: dm_channel)
+      message2.update(chat_channel: dm_channel)
+      message3.update(chat_channel: dm_channel)
```

Comment:
```suggestion
      message1.update!(chat_channel: dm_channel)
      message2.update!(chat_channel: dm_channel)
      message3.update!(chat_channel: dm_channel)
```

Reply:
Also, I'm not sure if we need to create 3 messages here since the messages are kind of irrelevant in this case.

Reply:
Yeah good point will just use one